modules/pipeline/spec/pipeline_extra.go

```diff
@@ -108,6 +108,8 @@ type PipelineExtraInfo struct {
 	QueueInfo *QueueInfo `json:"queueInfo,omitempty"`
 
 	TaskOperates []apistructs.PipelineTaskOperateRequest `json:"taskTaskOperates,omitempty"`
+
+	ContainerInstanceProvider *apistructs.ContainerInstanceProvider `json:"containerInstanceProvider,omitempty"`
```

Comment:
`ContainerInstanceProvider` is pipeline-level or task-level?
Can we support some tasks run on eci but others not on eci now?

Reply:
now `ContainerInstanceProvider` is pipeline-level, it will transfer to task

// todo task could custom eci, and pipeline server should judge the shard storage